openverse_catalog/dags/providers/provider_api_scripts/museum_victoria.py

```diff
@@ -1,203 +1,178 @@
-    if default_query_params is None:
-        default_query_params = DEFAULT_QUERY_PARAMS
-    query_params = default_query_params.copy()
-    query_params["imagelicence"] = license_type
-    query_params["page"] = page
-    return query_params
-
-
-def _get_batch_objects(endpoint=ENDPOINT, params=None, headers=None, retries=RETRIES):
-    if headers is None:
-        headers = HEADERS.copy()
-    data = None
-    for retry in range(retries):
-        response = delay_request.get(endpoint, params, headers=headers)
-        try:
-            response_json = response.json()
-            if type(response_json) == list:
-                data = response_json
-                break
-        except Exception:
-            data = None
-    return data
-
-
-def _handle_batch_objects(objects, landing_page=LANDING_PAGE):
-    image_count = 0
-    for obj in objects:
-        object_id = obj.get("id")
-        if object_id in RECORDS_IDS:
-            continue
-        RECORDS_IDS.append(object_id)
-        foreign_landing_url = landing_page + object_id
-        media_data = obj.get("media")
-        if media_data is None:
-            continue
-        image_data = _get_media_info(media_data)
-        if len(image_data) == 0:
-            continue
-        meta_data = _get_metadata(obj)
-        title = obj.get("displayTitle")
-        for img in image_data:
-            license_info = get_license_info(license_url=img.get("license_url"))
-            image_count = image_store.add_item(
-                foreign_identifier=img.get("image_id"),
-                foreign_landing_url=foreign_landing_url,
-                image_url=img.get("image_url"),
-                height=img.get("height"),
-                width=img.get("width"),
-                license_info=license_info,
-                title=title,
-                creator=img.get("creators"),
-                meta_data=meta_data,
-            )
-    return image_count
-
-
-def _get_media_info(media_data):
-    image_data = []
-    for media in media_data:
-        media_type = media.get("type")
-        if media_type == "image":
-            image_id = media.get("id")
-            image_url, height, width = _get_image_data(media)
-            license_url = _get_license_url(media)
-            if image_url is None or image_id is None or license_url is None:
+class ImageDetails(TypedDict, total=False):
+    foreign_identifier: str
+    image_url: str
+    license_info: LicenseInfo
+    foreign_landing_url: str
+    title: str
+    meta_data: Dict
+    height: int | None
+    width: int | None
+    creator: str | None
+
+
+class VictoriaDataIngester(ProviderDataIngester):
+    providers = {"image": prov.VICTORIA_DEFAULT_PROVIDER}
+    endpoint = "https://collections.museumsvictoria.com.au/api/search"
+    headers = {"User-Agent": prov.UA_STRING, "Accept": "application/json"}
+    batch_limit = 100
+    delay = 5
+    # This set is used to prevent duplicate images of the same items
+    RECORDS_IDS = set()
```

Comment:
Looking at this again, I think it may be ill-advised to have a mutable object attached to the class instance (same for #576) - do you think we could move this to an `__init__` override and set it once the instance is initialized instead?

Reply:
It turns out this is the _exact_ cause of the test failures! Multiple tests ingest that `record_data`, and so when more than one test is run the object already exists and isn't added 😅 Although even moving this into the `__init__` method wouldn't immediately fix this, we'd probably need to add a fixture called `mv` which returns a new `VictoriaDataIngester` instance for each test.

Reply:
Thank you so much for figuring this out, @AetherUnbound !!!
I've moved `RECORD_IDS` to `__init__`, and in the tests file added a fixture that runs after each test and replaces `RECORD_IDS` with a blank set.